pkg/grpc/universalapi/api_metadata.go

```diff
@@ -89,6 +89,11 @@ func (a *UniversalAPI) GetMetadata(ctx context.Context, in *emptypb.Empty) (*run
 
 // SetMetadata Sets value in extended metadata of the sidecar.
 func (a *UniversalAPI) SetMetadata(ctx context.Context, in *runtimev1pb.SetMetadataRequest) (*emptypb.Empty, error) {
+	// Nop if the key is empty
+	if in.Key == "" {
+		return &emptypb.Empty{}, nil
+	}
```

Comment:
please add a separate PR... Let this PR be only chi router change.

Reply:
I _think_ this was needed to fix a test that was failing. Due to how the new router works, it may have been needed.